prime-router/src/main/kotlin/Element.kt

```diff
@@ -575,7 +579,12 @@ data class Element(
                     if (nullifyValue) {
                         ""
                     } else {
-                        error("Invalid date: '$cleanedFormattedValue' for format '$format' for element $fieldMapping")
+                        throw ElementNormalizeException(
```

Comment:
We throw the new ElementNormalizeException in this class now so, when caught, we have all the info to make a nice action message with an accurate errorCode. Most of the changes in this file are to acomadate this concept.

FYI: error(...) ultimately throws an IllegalStateException.